### ISSUE TITLE
Sudden unavailable adressbook issue -  'Importing is disabled because there are no address books available' 

### ISSUE THREAD

**Issue by AKDub** [nextcloud/contacts#4223]
The address book I have been using seems to still be in the oc_adressbooks database.

I thought it might be related to this issue, but I'm not sure if my contacts are big enough - https://github.com/nextcloud/contacts/issues/2638

```MariaDB [nextcloud]> select id, LENGTH(carddata) FROM oc_cards ORDER BY LENGTH(carddata) DESC LIMIT 10;
+------+------------------+
| id   | LENGTH(carddata) |
+------+------------------+
| 3278 |           766995 |
| 3239 |           715150 |
| 3171 |           694025 |
| 3209 |           682165 |
| 2591 |           669057 |
| 3217 |           662580 |
| 3356 |           656131 |
| 2651 |           630007 |
| 3187 |           624511 |
| 3250 |           622130 |
+------+------------------+
10 rows in set (12.692 sec)```

Any ideas?

**Comment by AKDub** [comment]
Presume its one of these two 

`{"reqId":"pD2U6TdEOQ7fwiKABnmc","level":2,"time":"2024-11-26T16:30:08+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"Cannot retrieve value of undefined directive Core.RemoveBlanks invoked on line 75 in file /config/www/nextcloud/apps/news/vendor/ezyang/htmlpurifier/library/HTMLPurifier/Lexer/DOMLex.php at /config/www/nextcloud/apps/news/vendor/ezyang/htmlpurifier/library/HTMLPurifier/Config.php#901","userAgent":"--","version":"30.0.2.2","data":{"app":"PHP"}}
{"reqId":"SRqWhtzkRCKTdA1adPzV","level":3,"time":"2024-11-26T17:00:21+00:00","remoteAddr":"100.122.42.118","user":"jay","app":"PHP","method":"REPORT","url":"/remote.php/dav/addressbooks/users/jay/contacts/","message":"Allowed memory size of 536870912 bytes exhausted (tried to allocate 507904 bytes) at /config/www/nextcloud/apps/dav/lib/CardDAV/CardDavBackend.php#1009","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:132.0) Gecko/20100101 Firefox/132.0","version":"30.0.2.2","data":{"app":"PHP"}}`

**Comment by SebastianKrupinski** [comment]
@AKDub it seems you are running out of memory, we retrieving the contacts entries. How many contacts are in the addressbook?

**Comment by SebastianKrupinski** [comment]
I think it might be the photos causing it, at one point we had no limit on the size of the image, not any new image gets resized. But I believe any contact that is already existing with a image does not get resized.

**Comment by AKDub** [comment]
Added `memory_limit = 1024M` to the `php/php-local.ini` file, and and the contacts loaded up first time. No issue for me to add some RAM to the system, so I will just leave it like that.

Thanks you so much for the quick help today.

As an added bonus - the only thing that was in that file was `date.timezone =` and on a hunch I deleted that line and now I don't seem to have the annoying `NOTICE: PHP message: PHP Warning:  PHP Startup: Invalid date.timezone value '', using 'UTC' instead in Unknown on line 0` spamming the CLI.

Double-win!
